walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -35,22 +35,42 @@
       return ImmutableList.of();
     }
  
-    ImmutableSet<String> requiredAttendees = ImmutableSet.copyOf(request.getAttendees());
+    ImmutableList<String> requiredAttendees = ImmutableList.copyOf(request.getAttendees());
+    ImmutableList<String> optionalAttendees = ImmutableList.copyOf(request.getOptionalAttendees());
  
-    // Get time ranges from events that include at least one required attendee
-    ImmutableList<TimeRange> unavailableTimes = 
-        Streams.stream(events)
-            .filter(event -> !Collections.disjoint(requiredAttendees, event.getAttendees()))
-            .map(Event::getWhen)
-            .collect(toImmutableList());
+    // Get required and optional unavailable time ranges
+    ImmutableList<TimeRange> unavailableTimes = getUnavailableTimeRanges(events, requiredAttendees);
+    ImmutableList<TimeRange> optionalUnavailableTimes = getUnavailableTimeRanges(events, optionalAttendees);
  
     // No unavailable time ranges means the entire day is available
-    if (unavailableTimes.isEmpty()) {
+    if (unavailableTimes.isEmpty() && optionalUnavailableTimes.isEmpty()) {
       return ImmutableList.of(TimeRange.WHOLE_DAY);
     }
 
-    // TODO: Change structure to not update a class variable in the stream
     // Find all available time ranges for the requested meeting based on the attendees' unavailable times 
+    ImmutableList<TimeRange> availableTimes = ImmutableList.copyOf(getAvailableTimeRanges(unavailableTimes, request));
```

Comment:
It looks like you always convert the result of getAvailableTimeRanges to an ImmutableList?

If you do not expect to ever need to change the result of getAvailableTimeRanges I would change the method to return an ImmutableList<TimeRange> instead